### PR TITLE
Fix exception bug when there is no intersection of all the terms

### PR DESCRIPTION
**Pull request by Navnedia**
I found this bug with my implementation of the Dict index search, so I thought I would submit a pull request to fix it here too.

In the first loop where we find the intersected doc_ids, when we have a query where there is no intersection between all the terms, the empty set will get filled again in the next loop iteration it starts to build an intersection of the remaining terms. Once the bug intersection is finished it will throw an error when the score calculation loop tries to get the doc_id TF for a term it's not included in. I found this was an issue while running the evaluator with the term Dict index. Try searching query 4 with the Dict index to see what I mean.

### CODE REVIEW

File: index.py [hunk 1]
```diff
@@ -193,7 +193,7 @@ def read(self):
     def search(self, query: Query) -> SearchResults:
         result_doc_ids = None
         for term in query.terms:
-            if term not in self.term_to_doc_id_and_frequencies:
+            if term not in self.term_to_doc_id_and_frequencies or result_doc_ids == set():
                 return SearchResults([])
             if not result_doc_ids:
```
Comment (avasserman) [inline index.py:198]: Another way to fix this problem would be to replace 
            if not result_doc_ids:
with
            if result_doc_ids is None:
That was the intension. If result_doc_ids is empty, the current code reinitializes it, which was not the intension.

**Review by avasserman — APPROVED** [review]
Thank you for noticing and for the fix! Could you please also update line 196 as described in the comment.

File: index.py (hunk 1 above)
Comment (Navnedia) [reply index.py:198]: Are you wanting me to remove my solution and replace it with just that, or do you want me to add that in addition?

**Comment by avasserman** [comment]
Please add it to your fix. Early stopping you have done makes anyway.

**Comment by Navnedia** [comment]
I went ahead and made that requested change.

**Comment by avasserman** [comment]
Great, thanks!

**Review by avasserman — APPROVED** [review]

**Pull request merged**